File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        word_dic = collections.defaultdict(list)

        for s in strs:
            st = "".join(sorted(s))
            word_dic[st].append(s)
        return list(word_dic.values())
        

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();
     for (String st : strs){
        char[] chrs = st.toCharArray();
        Arrays.sort(chrs);
        String str_sorted = new String(chrs);
        if (!map.containsKey(str_sorted)){
            map.put(str_sorted, new ArrayList<>());
            
        }
        map.get(str_sorted).add(st);
        

     }  
    return new ArrayList<>(map.values()); 
    }
}